pkg/template/sops.go

```diff
@@ -206,37 +210,89 @@ func (*Sops) shapeDecryptedContents(contentsBs []byte) ([]byte, error) {
 	return contentsBs, nil
 }
 
-func (t *Sops) configPaths() (*memdir.TmpDir, string, error) {
-	gpgHomeDir := memdir.NewTmpDir("template-sops-config")
+type sopsCryptoStrategy int8 // just a hunch that we'll never have a 257th encryption provider but if you're still using this in the year 2666 and upgrading to an int16, sorry about this and also all the CO2, single-use plastic, and whatever else we did wrong
+const (
+	PGP sopsCryptoStrategy = iota
+	age
+)
 
-	err := gpgHomeDir.Create()
-	if err != nil {
-		return nil, "", err
+type sopsConfig struct {
+	cryptoHomeDir *memdir.TmpDir
+	configPath    string
+	agePublicKey  string
+	strategy      sopsCryptoStrategy
+}
+
+func (sc *sopsConfig) storeAgePublicKeyFrom(keyFile string) error {
+	myre := regexp.MustCompile(`(?:public key: )(.*)`)
+	matches := myre.FindStringSubmatch(keyFile)
```

Comment:
btw very open to other ideas re. how to get the public key --  we _could_ require it to be set in the resource itself? like:
```
sops:
  age: 
     secretRef:  whatever
     publicKey: age123213287439234
```

    but basically it's part of the second line of the keys file,  so i'm using regular expressions to scrape it out.